src/lib/sort/sort.ts

```diff
@@ -25,24 +26,25 @@ import {
   mixinDisabled,
   mixinInitialized,
 } from '@angular/material/core';
-import {Subject} from 'rxjs';
+import {Subject, Observable} from 'rxjs';
 import {SortDirection} from './sort-direction';
 import {
   getSortDuplicateSortableIdError,
   getSortHeaderMissingIdError,
   getSortInvalidDirectionError,
 } from './sort-errors';
+import {MatSortable, SortableState} from './sortable';
+import {MAT_SORT_CONTAINER} from './sort-module';
 
-/** Interface for a directive that holds sorting state consumed by `MatSortHeader`. */
-export interface MatSortable {
-  /** The id of the column being sorted. */
-  id: string;
-
-  /** Starting sort direction. */
-  start: 'asc' | 'desc';
+// Boilerplate for applying mixins to MatSort.
+/** @docs-private */
+export class MatSortBase {}
+export const _MatSortMixinBase: HasInitializedCtor & CanDisableCtor & typeof MatSortBase =
+    mixinInitialized(mixinDisabled(MatSortBase));
 
-  /** Whether to disable clearing the sorting state. */
-  disableClear: boolean;
+/** State provided to a `MatSortable`, includes the current `MatSort` active sortable. */
+export interface MatSortSortableState extends SortableState {
```

Comment:
This name is a bit...repetitive.

Reply:
I agree, but it may be more valuable to keep it description. In the future there will be things like `MatMultiSortSortableState`.

Options:
`MatSortableState`/`MatMultiSortableState`
`MatSortState`/`MatMultiSortState`